drivers/kscan/Kconfig

```diff
@@ -27,4 +27,77 @@ config KSCAN_INIT_PRIORITY
 	help
 	  Keyboard scan device driver initialization priority.
 
+config KSCAN_MATRIX
+	bool "keyboard scan matrix"
+	help
+	  A matrix circuit is used in most computer keyboards where the key
+	  switches are connected by a grid of wires, similar to a diode matrix.
+
+if KSCAN_MATRIX
+config KSCAN_MATRIX_MAX_COLUMNS
+	int "keyboard matrix max column"
+	default 18
+	help
+	  Maximum columns of the matrix circuit.
+
+config KSCAN_MATRIX_MAX_ROWS
+	int "keyboard matrix max row"
+	default 8
+	help
+	  Maximum rows of the matrix circuit.
+
+config KSCAN_MATRIX_DRIVER
```

Comment:
Move the KSCAN_MATRIX_DRIVER configs into a separate file.

Reply:
I moved the whole KSCAN_MATRIX configs into Kconfig.matrix file.